### PR TITLE
feat: initial changes for adoption of file dataType

### PR DESCRIPTION
**Pull request by Shurtu-gal**
## PR Details

- I have added `File` in the data-types enum.
- Also added the necessary schema for it.

## Screenshots
image
image

## PR Checklist

- [x] Add schema for file data type
- [x] Check if the client is properly rendering the fields
- [ ] Add plugin checking in DSG
- [ ] Tests for the changes have been added
- [ ] `npm test` doesn't throw any error

IMPORTANT: Please review the [CONTRIBUTING.md](https://github.com/amplication/amplication/blob/master/CODE_OF_CONDUCT.md) file for detailed contributing guidelines.

### CODE REVIEW

File: packages/amplication-client/src/Entity/SchemaField.tsx [hunk 1]
```diff
@@ -36,26 +36,33 @@ export const SchemaField = ({
...
 
-  const enumOptions = useMemo((): OptionItem[] | null => {
-    if (propertySchema.enum) {
-      return (propertySchema.enum as string[]).map((item) => {
-        const labelByItem: { label: string; value: string } =
-          ENTITY_FIELD_ENUM_MAPPER[fieldDataType][propertyName].find(
-            (option) => option.value === item
-          );
+  const enumOptions = useCallback(
+    function (enumOptions: string[]): OptionItem[] | null {
```
Comment (overbit) [inline packages/amplication-client/src/Entity/SchemaField.tsx:40]: Let's try to be consistent with the current code that is using lambda functions

File: packages/amplication-client/src/Entity/SchemaField.tsx [hunk 2]
```diff
@@ -36,26 +36,33 @@ export const SchemaField = ({
...
+  );
 
   if (propertySchema.enum) {
     if (propertySchema.enum.every((item) => typeof item === "string")) {
       return (
-        <SelectField label={label} name={fieldName} options={enumOptions} />
+        <SelectField
+          label={label}
+          name={fieldName}
+          options={enumOptions(propertySchema.enum as string[])}
```
Comment (overbit) [inline packages/amplication-client/src/Entity/SchemaField.tsx:64]: Is this casting necessary? `typeof` should already instruct TS compiler about the type.

File: packages/amplication-client/src/Entity/SchemaField.tsx [hunk 3]
```diff
@@ -36,26 +36,33 @@ export const SchemaField = ({
...
   const label = propertySchema.title || capitalCase(propertyName);
 
-  const enumOptions = useMemo((): OptionItem[] | null => {
-    if (propertySchema.enum) {
-      return (propertySchema.enum as string[]).map((item) => {
-        const labelByItem: { label: string; value: string } =
-          ENTITY_FIELD_ENUM_MAPPER[fieldDataType][propertyName].find(
-            (option) => option.value === item
-          );
+  const enumOptions = useCallback(
```
Comment (overbit) [inline packages/amplication-client/src/Entity/SchemaField.tsx:39]: As the `propertySchema.enum`, can we keep the `useMemo`?

File: packages/amplication-code-gen-types/src/schemas/file.json [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+      "additionalItems": true,
+      "minItems": 0,
+      "items": {
+        "$comment": "String that the MIME type of the file must contain or a option from the enum",
+        "anyOf": [
+          {
+            "type": "string"
+          },
+          {
+            "enum": ["image/", "audio/", "video/", "application/", "document/"]
```
Comment (overbit) [inline packages/amplication-code-gen-types/src/schemas/file.json:23]: As previously discussed, we should either leave the `allowedMimeTypes` open or add all possible options currently registered as IANA.
The following types are missing:
- `font/`
- `model/`
- `example/`
- `text/`

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-code-gen-types/src/schemas/file.json (hunk 4 above)
Comment (Shurtu-gal) [reply packages/amplication-code-gen-types/src/schemas/file.json:23]: It is open (you can add any arbitrary string), the enum is just a way to make it easy for the users to add the common mime types, meanwhile, I will add these as well.

**Review by overbit — CHANGES_REQUESTED** [review]

File: packages/amplication-client/src/Entity/SchemaField.tsx (hunk 3 above)
Comment (Shurtu-gal) [reply packages/amplication-client/src/Entity/SchemaField.tsx:39]: Nevermind, as suggested I am going with the arbitrary array of strings approach for now.

File: packages/amplication-code-gen-types/src/schemas/file.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
+{
+  "$id": "https://amplication.com/schema/entityfield/properties/file.json",
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "title": "file",
+  "type": "object",
+  "additionalProperties": false,
+  "properties": {
```
[the remark this answers is not in the document]
Comment (Shurtu-gal) [reply packages/amplication-code-gen-types/src/schemas/file.json:7]: Sorry, was overlooked by me, have added it.

**Review by overbit — APPROVED** [review]

**Pull request merged**